src/ShaderTools.CodeAnalysis.Hlsl/Syntax/SyntaxRewriter.cs

```diff
@@ -8,17 +9,60 @@ public abstract partial class SyntaxRewriter : SyntaxVisitor<SyntaxNode>
     {
         public List<TNode> VisitList<TNode>(List<TNode> nodeList) where TNode : SyntaxNode
         {
-            throw new NotImplementedException();
+            List<TNode> newNodes = null;
+            for (var i = 0; i < nodeList.Count; i++)
+            {
+                var n = nodeList[i];
+                var nt = n as SyntaxToken;
+                var nn = nt == null ? n.Accept(this) : this.VisitToken(nt);
```

Comment:
We need an exception for SyntaxToken because VisitSyntaxToken needs to returns a SyntaxToken instead of a SyntaxNode.

Roslyn uses a special type for lists of syntax tokens so they can have an overload for them instead of type checking like I did ([link](https://github.com/dotnet/roslyn/blob/2228cce387e5fd7739d8f9bbb713cf50a0604c06/src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxRewriter.cs#L246)). 
We can't use the Accept method for syntax tokens because the visit method needs